dockstore-webservice/src/main/java/io/dockstore/webservice/core/Collection.java

```diff
@@ -142,6 +141,11 @@ public void setDescription(String description) {
         return entries.stream().filter(entry -> entry.getIsPublished()).collect(Collectors.toSet());
     }
 
+    @JsonProperty("entries")
+    public Set<CollectionEntry> getCollectionEntries() {
+        return getEntries().stream().map(entry -> new CollectionEntry(entry)).collect(Collectors.toSet());
```

Comment:
This does reduce response size but `getEntries` will still load all entries (and thus their versions) into memory.
Is there a reason the named query strategy was not used in this case? 
https://github.com/dockstore/dockstore/pull/3462/files#diff-f57cce37c04c1e6fa1392a205af9efefR41

Reply:
Require DAO access which we normally do in a specific endpoint.  This is a JsonProperty.

Reply:
Yes, I think you will need to modify the DAO and tackle specific endpoints to avoid loading entries and their versions and avert OOM
At least until #3455

Reply:
that would be a relatively major change for a hotfix, modifying almost every endpoint that uses collection, ideally creating another endpoint to specifically get entries, and major modifications to the frontend entries retrieval code.